client/data/home/use-home-layout-query.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { useQuery, UseQueryResult } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+import { getHomeLayoutFlags } from './home-layout-flags';
+
+interface Options {
+	fetchOnMount?: boolean;
+}
+
+const useHomeLayoutQuery = (
+	siteId: number | null,
+	{ fetchOnMount = false }: Options = {}
+): UseQueryResult => {
+	const { isDev, forcedView } = getHomeLayoutFlags();
+
+	return useQuery(
+		[ 'home-layout', siteId ],
+		() =>
+			wpcom.req.get(
+				{
+					path: `/sites/${ siteId }/home/layout`,
+					apiNamespace: 'wpcom/v2',
+				},
+				{
+					...( isDev && { dev: true } ),
+					...( forcedView && { view: forcedView } ),
+				}
+			),
+		{
+			enabled: !! siteId,
+
+			// The `/layout` endpoint can return a random view. Disable implicit refetches
+			// so the view doesn't change without some user action.
+			staleTime: Infinity,
+			refetchOnMount: fetchOnMount && 'always',
```

Comment:
The original `<QueryHomeLayout/>` always fetches on mount and only on mount. To reproduce that, `refetchOnMount` should be always `always`. You might need to set `refetchInterval: false`, too.

Reply:
> The original `<QueryHomeLayout/>` always fetches on mount and only on mount.

That's true, but we also only included `QueryHomeLayout` in one place. The `useHomeLayoutQuery` hook is included in 3 places and I only want the request to happen once (always using `always` results in two requests, I think the 2nd two mounts are merged into one request because they mount in the same render pass).

I'm replicating the old behaviour by having the component that used to include `<QueryHomeLayout/>` now explicitly saying `fetchOnMount=true`, and the other places that used to just select the layout data leave `fetchOnMount` as `false`.

> You might need to set `refetchInterval: false`, too.

`false` is the default, but it wouldn't hurt to be explicit in case the default ever changes. I am trying to say "don't leave it up to React Query to do what it thinks is best" after all.

Reply:
> The `useHomeLayoutQuery` hook is included in 3 places and I only want the request to happen once

I see, the other two places are the ones that used to just passively call the selector. The best way to make a passive React Query like this is probably:
```js
useHomeLayoutQuery( siteId, { enabled: false } );
```
This hook will only read the current cached state and will not actively trigger anything.

I think it's better because it disables all fetches and retries with just one flag, instead of carefully disabling all methods (on mount, on window focus, on interval, ...) one by one.

Reply:
I've switched the `useHomeLayoutQuery()` hook over from not querying by default, to querying by default. It feels like our custom queries should have some sort of convention when a situation like this arises, and I like the idea of forwarding an `enable` option to `useQuery()`. Using the same flags as ReactQuery is a good convention.